internal/server/monitoring.go

```diff
@@ -24,16 +24,32 @@ func getMonitoringIntervalSeconds() time.Duration {
 func startGraphSizeMonitoring(interval time.Duration, database knowledge.GraphDB, sourcesRegistry sources.Registry) {
 	logrus.Infof("Monitoring of the graph size will happen every %ds", int(interval/time.Second))
 	go func() {
-		ctx, cancel := context.WithTimeout(context.Background(), interval)
-		defer cancel()
-
 		for {
+			ctx, cancel := context.WithTimeout(context.Background(), interval)
+			defer cancel()
```

Comment:
careful, you pilling up defer calls here, this will blow up the stack if I'm not mistaken

Reply:
I was thinking that the defer would free its stack frame at the end of the closure... No?

Reply:
You're right... https://stackoverflow.com/a/45617791